src/numeric/processes.jl

```diff
@@ -110,116 +129,59 @@ end
-      # Discretizing Exo process usinf QR
-      #This function computes the points and weights of an N-point Gauss–Legendre quadrature rule on the interval (a,b).
-      #It uses the O(N2) algorithm described in Trefethen & Bau, Numerical Linear Algebra, which
-      # finds the points and weights by computing the eigenvalues and eigenvectors of a real-symmetric tridiagonal matrix:
-      function gauss(a, b, S)
-          λ, Q = eig(SymTridiagonal(zeros(S), [ n / sqrt(4n^2 - 1) for n = 1:S-1 ]))
-          return (λ + 1) * (b-a)/2 + a, [ 2*Q[1, i]^2 for i = 1:S ] * (b-a)/2
-      end
-
-      x = collect(zeros(n_integration[ii]) for ii in 1:n);
-      w = collect(zeros(n_integration[ii]) for ii in 1:n);
-      for j in 1:n
-          x[j], w[j] = gauss(-VAR_info.Sigma[j, j], VAR_info.Sigma[j, j], n_integration[j])
-      end
-
-      # Computing the nodes of the exo process
-      Eⁱʲ=collect(zeros(n_integration[ii]) for ii in 1:n);
-      #Compute states of exogenous process
-      # j stands for a variables in the VAR
-
-      [Eⁱʲ[j] =  1/(2*VAR_info.Sigma[j, j]*sqrt(pi)).*exp(-(x[j]- μ).^2./(2*VAR_info.Sigma[j, j]^2))  for j = 1:n]
-      #Compute Cartesian product
-
-      Eⁱʲ_c = collect(Base.product([Eⁱʲ[i] for i in 1:size(Eⁱʲ, 1)]...))
-      #Compute future states for endogenous variables
-      # kron([n_integration[i] for i in 1:n]...) computes the possible combination of states of innovations
-      if n>2
-        yⁱʲ  = zeros(kron([n_integration[i] for i in 1:n]...), n, kron([n_states[i] for i in 1:n]...));
-
-        for ii in 1:kron([n_states[i] for i in 1:n]...)
-
-            [ yⁱʲ[ss, :, ii] = VAR_info.M + VAR_info.R*collect(yⁱ[ii])+collect(Eⁱʲ_c[ss]) for ss = 1:kron([n_integration[i] for i in 1:n]...) ]
-
-        end
-        integration_nodes = collect(yⁱʲ[:, :, ii] for ii in 1:kron([n_states[i] for i in 1:n]...))
-
-        # Get transition probabilities
-
-        integration_weights = repmat(kron([w[i] for i in 1:n]...), 1, kron([n_integration[i] for i in 1:n]...))
-        integration_weights =collect(integration_weights[:, ii] for ii in 1:kron([n_integration[i] for i in 1:n]...))
-      else
-        yⁱʲ  = zeros(n_integration[n], n, n_states[n]);
-
-        for ii in 1:n_states[n]
-            [ yⁱʲ[ss, :, ii] = VAR_info.M + VAR_info.R*collect(yⁱ[ii])+collect(Eⁱʲ_c[ss]) for ss = 1:n_integration[n]]
-
-        end
-        integration_nodes = collect(yⁱʲ[:, :, ii] for ii in 1:n_states[n])
-
-        # Get transition probabilities
-
-        integration_weights = w[n]
-        integration_weights =[integration_weights]
-      end
-#  return nodes, integration_nodes, integration_weights
-    return DiscretizedProcess(nodes, integration_nodes, integration_weights)
+function discretize(var::VAR1, n_states::Array{Int64,1}, n_integration::Array{Int64,1}; n_std::Int64=2,)
+
+    R = var.R
+    M = var.M
+    Sigma = var.Sigma
+    S = QE.solve_discrete_lyapunov(R,Sigma)  # asymptotic variance
```

Comment:
yeah, that one also fails for degenerate Sigma, so I have also considered rewriting it. Maybe I'll just open an issue against QuantEcon.